File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/repositories/tenant/OauthClientRepository.java
package com.bittokazi.oauth2.auth.server.app.repositories.tenant;

import com.bittokazi.oauth2.auth.server.app.models.tenant.OauthClient;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface OauthClientRepository extends JpaRepository<OauthClient, String> {

    @Query(value = "SELECT * FROM oauth_client_details WHERE id = :id", nativeQuery = true)
    Optional<OauthClient> findOneById(String id);

    @Query(value = "SELECT * FROM oauth_client_details WHERE client_id = :clientId", nativeQuery = true)
    Optional<OauthClient> findOneByClientId(String clientId);

}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/repositories/tenant/OauthClientRepository.kt
package com.bittokazi.oauth2.auth.server.app.repositories.tenant

import com.bittokazi.oauth2.auth.server.app.models.tenant.OauthClient
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface OauthClientRepository : JpaRepository<OauthClient, String> {
    @Query(value = "SELECT * FROM oauth_client_details WHERE id = :id", nativeQuery = true)
    fun findOneById(id: String?): Optional<OauthClient>

    @Query(value = "SELECT * FROM oauth_client_details WHERE client_id = :clientId", nativeQuery = true)
    fun findOneByClientId(clientId: String?): Optional<OauthClient>
}
